### PR TITLE
The list of tests should include TestNG tests.

### PR DESCRIPTION
**Pull request by lahodaj**
Please see here:
https://github.com/oracle/javavscode/issues/28

The test indexer uses providers to detect tests, and the provider for TestNG tests seems to be missing. So, this PR adds the provider.

### CODE REVIEW

**Comment by robaho** [comment]
Can this PR be approved and merged? This is holding up core functionality in the vscode extension based on the NetBeans language server.

**Review by dbalek — APPROVED** [review]

**Comment by neilcsmith-net** [comment]
Which milestone are you targetting with this?  Please add.  Could be a fix for release?

**Comment by robaho** [comment]
This fix is needed to make the Oracle VSCode extension work for a large number of developers. Thanks for your help.

**Comment by lahodaj** [comment]
@neilcsmith-net, while I think this is fairly safe, not sure if it is necessary to rush this into 20. So, I was more expecting this to be added to master, and released as part of a future release. The Java Platform Support extension can use this fix as a patch.

I don't object against this being added to 20, though, if there's enough time for it.

**Comment by neilcsmith-net** [comment]
@lahodaj there will be at least one more release candidate (probably more as there are a few open issues).  Up to you whether you think it warrants going in.  Wasn't sure if you also need it in a release for your usage?

Either put NB20 milestone and rebase to delivery, or put NB21 on it.

@robaho our primary concern is the large number of NetBeans IDE users.  The needs of other projects are secondary.

**Comment by robaho** [comment]
Makes sense. Still I would think it’s a pretty big issue for net brand users as well. The OpenJDK uses a lot of TrstNG tests.

**Comment by arvindaprameya** [comment]
Thank you for doing this, looks good

**Review by mbien — APPROVED** [review]
code looks good and is self contained which should make it low risk for last minute inclusion for RC3.

PR needs to be rebased to the delivery branch soon or it will miss the deadline.

Although this isn't tested directly it seems, we should run at least the java tests before integration -> I added the https://github.com/apache/netbeans/labels/Java label which will activate them on next sync.

**Comment by neilcsmith-net** [comment]
Added `do not merge` while branch decision / rebase decision made so we (well, I) don't accidentally merge in the wrong place now it's got the milestone on it.

**Comment by lahodaj** [comment]
If there are no objections, I'll merge tomorrow.

**Comment by mbien** [comment]
before pressing merge, please check if master is green via the actions tab: https://github.com/apache/netbeans/actions?query=event%3Apush+branch%3Amaster

I am having problems to get green results for micronaut and lsp jobs ATM - can't quite explain why, since the last integrated PR should only concern PHP.

**Comment by mbien** [comment]
#6673 proposes a fix to the test issues. However this will have to go through delivery first if there is another RC, so it might take a bit longer until master is stable.

**Comment by neilcsmith-net** [comment]
@mbien master is up-to-date with delivery already, so when the fix is ready to merge I'll immediately sync just that to master rather than wait for anything else.

**Comment by lahodaj** [comment]
Master seems to be green, merging...

**Pull request merged**
